node/core/prospective-parachains/src/fragment_tree.rs

```diff
@@ -53,6 +58,55 @@
 //! Our assumption is that the amount of candidates and parachains we consider will be reasonably
 //! bounded and in practice will not exceed a few thousand at any time. This naive implementation
 //! will still perform fairly well under these conditions, despite being somewhat wasteful of memory.
+//!
+//! # Main Types
+//!
+//! - [`FragmentTree`]
+//!   - Created with a given scope and populated from storage.
+//!     - [`FragmentTree::populate`]
+//!     - Can be populated recursively (i.e. `populate` will pick up
```

Comment:
Worth noting that it's technically possible for cycles to occur, if the PVF is crafted in such a way that cycles are possible.

Reply:
I have a section about cycles a bit further down. Added a note.